### DISCUSSION TITLE
GPU Acceleration?

### DISCUSSION THREAD

**Discussion by BergChristian** [saucecontrol/PhotoSauce#85] [Q&A]
Hi!

Love your library. It is very fast and just works very good! I ran into this project on codeproject and thought that maybe that could help you getting even high performance? 
https://www.codeproject.com/Articles/1116907/How-to-Use-Your-GPU-in-NET
And by offloading the CPU it will save some CPU for other stuff.

Just an idea, thank you for the work you have done.

Christian

**Comment by saucecontrol** [comment]
Hi Christian, I appreciate the kind words!

GPU acceleration isn't currently on the roadmap, for a couple of reasons.

1) In order to overcome the cost of copying data back and forth from the GPU, you have to do a large amount of computation on that data.  There is no processing in the MagicScaler pipeline that currently meets the bar where this would be profitable, so there isn't anything to be gained -- at least not with the current feature set.
2) The primary use case for the MagicScaler pipeline is high-volume image processing in a server environment.  Since outside of specialized (and expensive) ML environments, GPUs are not typically available on servers, it makes more sense to concentrate on making CPU-only processing as fast as possible.

Don't worry, the library will continue to get faster anyway.  The current bottlenecks are mostly in the image encoders and decoders, so those are the next targets 😃

**Reply by BergChristian** [reply]
Thank you for the answer! 

Fully understand that, I have the same discussions for what we do internally in the software. Is it worth sending to graphics card? and it is quite often the case that it is not worth it.

Keep up the good work!

All the best
Christian
